hashring/iter.go

```diff
@@ -20,146 +20,81 @@
 
 package hashring
 
-//Iter returns an iterator starting at the leftmost node in the tree
-func (t *RBTree) Iter() *RBIter {
-	return NewRBIter(t)
+// Iterator is used to perform an in-order traversal of the Red-Black Tree.
+// Repedetly call Next() on this to receive new
+type Iterator struct {
+	itFunc iteratorFunc
 }
 
-// IterAt returns an iter set at, or immediately after, val
-func (t *RBTree) IterAt(val int) *RBIter {
-	return NewRBIterAt(t, val)
+// NewIterator instantiates and returns a new Iterator that iterates over
+// a red black from the first node to the last.
+func NewIterator(tree *RedBlackTree) *Iterator {
+	return &Iterator{iterate(tree.root)}
 }
 
-// An RBIter iterates over nodes in an RBTree
-type RBIter struct {
-	tree      *RBTree
-	current   *RBNode
-	ancestors []*RBNode
+// NewIteratorAt instantiates and returns a new Iterator that iterates over
+// a red black from the node that owns the given value v to the last.
+func NewIteratorAt(tree *RedBlackTree, v int) *Iterator {
```

Comment:
Why is this now a function rather than a method, given that it takes a tree arg?